### PR TITLE
Keep unit selections when switching game mode or menu

### LINKED ISSUE #573: Keep unit selections when switching game mode or menu
Don't loose unit selections when the game mode is switched. Later, e.g. when opening a chat window or pressing escape, this will become important.

Currently, the selection is lost when switching game mode.

This is a follow up for #572.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/35009187-keep-unit-selections-when-switching-game-mode-or-menu?utm_campaign=plugin&utm_content=tracker%2F371436&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F371436&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by kev946**
Fixes #573

### CODE REVIEW

File: libopenage/engine.h [hunk 1]
```diff
@@ -411,6 +414,8 @@ class Engine : public ResizeHandler, public options::OptionNode {
 	 */
 	input::InputManager input_manager;
 
+	std::unique_ptr<UnitSelection> unit_selection;
```
Comment (TheJJ) [inline libopenage/engine.h]: could you please add documentation here and for the getter function above?

**Comment by TheJJ** [comment]
looks good otherwise :) you need to fix your merge conflicts though and then rebase please.

**Comment by kev946** [comment]
@TheJJ Rebased and fixed merge conflicts

**Comment by TheJJ** [comment]
thx :dancers:

**Pull request merged**